shopinvader/services/cart.py

```diff
@@ -278,6 +335,7 @@ def _prepare_cart(self):
             ] = self.shopinvader_backend.account_analytic_id.id
         if self.shopinvader_backend.sequence_id:
             vals["name"] = self.shopinvader_backend.sequence_id._next()
+        vals.update({"pricelist_id": self.shopinvader_backend.pricelist_id.id})
```

Comment:
@acsonefho Can you explain why we need to add the pricelist here?

Reply:
Because the pricelist used is the one define on the partner_id (of the SO by default). So the pricelist define on the backend was completely ignored (so my new unit test from yesterday) didn't work due to this mistake.

Reply:
@acsonefho Can you add this fix and the related tests into a separate commit PLZ